File: src/main/java/org/game/Medium.java
package org.game;

import javax.swing.*;

/**
 * 16 x 16, 64 mines
 */

public final class Medium implements GameType {
    private Medium() { }

    /**
     * 16 x 16, 64 mines
     * @param name name of player
     */

    public static final void start(final String name, final String password) {
        JOptionPane.showMessageDialog(
                null,
                "Good luck, " + name + "!",
                "Greetings",
                JOptionPane.INFORMATION_MESSAGE
        );

        new Game(16, 16, 64, name, password, new Medium()).start();
    }
}

File: src/main/kotlin/org/game/Medium.kt
package org.game

import javax.swing.JOptionPane

/**
 * 16 x 16, 64 mines
 */


internal class Medium : GameType {
    companion object {

        /**
         * 16 x 16, 64 mines
         * @param name name of player
         */

        fun start(name: String, password: String) {
            JOptionPane.showMessageDialog(
                null,
                "Good luck, $name!",
                "Greetings",
                JOptionPane.INFORMATION_MESSAGE
            )

            Game(16, 16, 64, name, password, Medium()).start()
        }
    }
}